actionpack/lib/action_dispatch/middleware/cookies.rb

```diff
@@ -389,6 +395,7 @@ def []=(name, options)
           options = { value: value }
         end
 
+        preprocess_options(options)
         handle_options(options)
```

Comment:
Of course, if you know why it isn't necessary for `delete` and `deleted?` do explain and we'll just leave this as is.